templates/webhook.bin.erb

```diff
@@ -108,7 +114,13 @@ class Server < Sinatra::Base
     data = JSON.parse(decoded, :quirks_mode => true)
 
     # github sends a 'ref', stash sends an array in 'refChanges'
-    branch = ( data['ref'] || data['refChanges'][0]['refId'] ).sub('refs/heads/', '')
+    if $config['git_provider'] == 'github'
+      branch = ( data['ref'] || data['refChanges'][0]['refId'] ).sub('refs/heads/', '')
```

Comment:
So Basically I would want to make this a required option for all services or fully automatic like it is not, My assumtion is you could just move your code `data['pullrequest_merged']['destination']['branch']['name']` to the end of the `||` conditionals and this would work out of the box without the need for the param to exist.